### PR TITLE
Mention 'BuildLayout' in section on available services

### PR DESCRIPTION
**Pull request by abstratt**
Issue: #16506

### CODE REVIEW

**Comment by abstratt** [comment]
@bot-gradle help

**Comment by abstratt** [comment]
@bot-gradle BD

**Comment by abstratt** [comment]
Sample here: https://builds.gradle.org/repository/download/Gradle_Release_Check_BuildDistributions/75121412:id/distributions/gradle-8.5-docs.zip!/gradle-8.5-20231128153656%2B0000/docs/userguide/custom_gradle_types.html#service_injection

File: subprojects/docs/src/docs/userguide/extending-gradle/custom_gradle_types.adoc [hunk 1]
```diff
@@ -187,6 +187,7 @@ The following services are available for injection:
 - link:{javadocPath}/org/gradle/api/file/ArchiveOperations.html[ArchiveOperations] - Allows a task to run operations on archive files such as ZIP or TAR files.
 - link:{javadocPath}/org/gradle/process/ExecOperations.html[ExecOperations] - Allows a task to run external processes with dedicated support for running external `java` programs.
 - link:{javadocPath}/org/gradle/tooling/provider/model/ToolingModelBuilderRegistry.html[ToolingModelBuilderRegistry] - Allows a plugin to registers a Gradle tooling API model.
+- link:{groovyDslPath}/org.gradle.api.file.BuildLayout.html[BuildLayout] - Provides access to important locations for a Gradle build. This service is only available for injection in settings plugins.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/extending-gradle/custom_gradle_types.adoc:190]: 🤔 Would it make sense to list it next to (before or after) `ProjectLayout` ?

**Review by ljacomet — APPROVED** [review]
LGTM, thanks for adding this quickly

One minor note below. Happy with you taking it into account or ignoring.

**Pull request merged**